### PR TITLE
layers: Simplify expressions

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Running code analysis, many expressions could be simplified

### CODE REVIEW

File: layers/gpu_validation/debug_printf.cpp [hunk 1]
```diff
@@ -36,11 +36,11 @@ void DebugPrintf::CreateDevice(const VkDeviceCreateInfo *pCreateInfo) {
 
     std::string verbose_string = getLayerOption("khronos_validation.printf_verbose");
     transform(verbose_string.begin(), verbose_string.end(), verbose_string.begin(), ::tolower);
-    verbose = verbose_string.length() ? !verbose_string.compare("true") : false;
+    verbose = verbose_string.empty() && !verbose_string.compare("true");
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/gpu_validation/debug_printf.cpp:39]: sorry, I think this should be

```patch
- verbose = verbose_string.empty() && !verbose_string.compare("true");
- verbose = !verbose_string.empty() && !verbose_string.compare("true");
```

[the remark this answers is not in the document]
Comment (artem-lunarg) [reply layers/gpu_validation/debug_printf.cpp]: Yes, this looks like an equivalent transformation. Still check for empty() looks like unnecessary call, iirc compare() takes into account length.

[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/gpu_validation/debug_printf.cpp]: >  It's probably enough just to write verbose = verbose_string.compare("true") == 0 without empty check

I see, good call, will fix

**Review by artem-lunarg — APPROVED** [review]

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
Fix commit typo

**Pull request merged**